apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/payloads_decoder.ex

```diff
@@ -65,8 +66,22 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.PayloadsDecoder do
   Safely decodes a zlib deflated binary and inflates it.
   This function avoids zip bomb vulnerabilities, and it decodes up to 10_485_760 bytes.
   """
-  @spec safe_inflate(binary) :: binary
+  @spec safe_inflate(binary) :: binary | :error
   def safe_inflate(zlib_payload) do
+    try do
```

Comment:
Elixir supports `catch` at the `def` level (see [here](https://hexdocs.pm/elixir/1.16/Kernel.html#def/2-rescue-catch-after-else)), so you can skip the `do_` helper and just do

```elixir
def safe_inflate(zlib_payload) do
  ...
catch
  :error, error ->
    ...
end